src/collection-preferences/content-display/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { DndContext } from '@dnd-kit/core';
+import { arrayMove, SortableContext, verticalListSortingStrategy } from '@dnd-kit/sortable';
+import { SortableItem } from './sortable-item';
+import useDragAndDropReorder from './use-drag-and-drop-reorder';
+import useLiveAnnouncements from './use-live-announcements';
+
+const componentPrefix = 'content-display';
+
+const getClassName = (suffix: string) => styles[`${componentPrefix}-${suffix}`];
+
+interface ContentDisplayPreferenceProps extends CollectionPreferencesProps.ContentDisplayPreference {
+  onChange: (value: ReadonlyArray<CollectionPreferencesProps.ContentDisplayItem>) => void;
+  value?: ReadonlyArray<CollectionPreferencesProps.ContentDisplayItem>;
+}
+
+export default function ContentDisplayPreference({
+  title,
+  description,
+  options,
+  value = options.map(({ id }) => ({
+    id,
+    visible: true,
+  })),
+  onChange,
+  liveAnnouncementDndStarted,
+  liveAnnouncementDndItemReordered,
+  liveAnnouncementDndItemCommitted,
+  liveAnnouncementDndDiscarded,
+  dragHandleAriaDescription,
+  dragHandleAriaLabel,
+}: ContentDisplayPreferenceProps) {
+  const idPrefix = useUniqueId(componentPrefix);
+
+  const onToggle = (option: OptionWithVisibility) => {
+    onChange(value.map(item => (item.id === option.id ? { ...item, visible: !option.visible } : item)));
+  };
+
+  const titleId = `${idPrefix}-title`;
+  const descriptionId = `${idPrefix}-description`;
+
+  const sortedOptions = getSortedOptions({ options, contentDisplay: value });
+
+  const { activeItem, collisionDetection, handleKeyDown, isKeyboard, sensors, setActiveItem } = useDragAndDropReorder({
+    sortedOptions,
+  });
+
+  const announcements = useLiveAnnouncements({
+    isDragging: activeItem !== null,
+    liveAnnouncementDndStarted,
+    liveAnnouncementDndItemReordered,
+    liveAnnouncementDndItemCommitted,
+    liveAnnouncementDndDiscarded,
+    sortedOptions: value,
+  });
+
+  return (
+    <div className={styles[componentPrefix]}>
+      <h3 className={getClassName('title')} id={titleId}>
+        {title}
+      </h3>
+      <p className={getClassName('description')} id={descriptionId}>
+        {description}
+      </p>
+      <DndContext
+        sensors={sensors}
+        collisionDetection={collisionDetection}
+        accessibility={{
+          announcements,
+          restoreFocus: false,
+          screenReaderInstructions: dragHandleAriaDescription ? { draggable: dragHandleAriaDescription } : undefined,
+        }}
+        onDragStart={({ active }) => setActiveItem(active.id)}
+        onDragEnd={event => {
+          setActiveItem(null);
+          const { active, over } = event;
+
+          if (over && active.id !== over.id) {
+            const oldIndex = value.findIndex(({ id }) => id === active.id);
+            const newIndex = value.findIndex(({ id }) => id === over.id);
+            onChange(arrayMove([...value], oldIndex, newIndex));
```

Comment:
Should we check if oldIndex != newIndex?

Reply:
We're already checking that `active.id !== over.id` above, so `active.id` and `over.id` should be in different positions and therefore `oldIndex` and `newIndex` will be different.